vnext/Microsoft.ReactNative/ABICxxModule.cpp

```diff
@@ -45,6 +45,15 @@ std::map<std::string, folly::dynamic> ABICxxModule::getConstants() noexcept {
 
 std::vector<CxxModule::Method> ABICxxModule::getMethods() noexcept {
   auto result = std::move(m_methods);
+  if (result.empty()) {
+    // Module without methods are not registered in JS configuration.
+    // It causes index mismatch between JS and C++ code.
+    // As a result, JS code will call methods on a wrong native module.
+    // See for details ReactCommon\cxxreact\ModuleRegistry.cpp, line 133: that reads 'if (!methodNames.empty()) {'
```

Comment:
>line 133: that reads 'if (!methodNames.empty() [](start = 64, length = 46)

Thanks for fixing this!  Question:  should we also upstream a change to facebook to remove this check from ModuleRegistry.cpp (or some other way to guarantee we don't corrupt the module table)?

Reply:
Thank you for approving the PR! Yes, I think we should try to upstream it. Though it is probably a low priority.

Reply:
I have got to the root of the issue: instead of the previous work around with adding a Dummy method, I found that the issue is in our OInstance.cpp. I made it to match the iOS ReactNative code in the RCTObjExecutor.mm to avoid index mismatch.